lib/semian/mysql2.rb

```diff
@@ -53,11 +55,19 @@ def semian_identifier
     end
 
     def query(*args)
-      acquire_semian_resource(adapter: :mysql, scope: :query) { raw_query(*args) }
+      if query_whitelisted?(*args)
+        raw_query(*args)
+      else
+        acquire_semian_resource(adapter: :mysql, scope: :query) { raw_query(*args) }
+      end
     end
 
     private
 
+    def query_whitelisted?(sql, *)
+      QUERY_WHITELIST.any? { |pattern| sql =~ pattern }
```

Comment:
> how long the actual query will take.

That does not matter the queries will be as fast as they were before or after this patch. But it seems that regexing will consume overall more CPU if you say do many queries on 100s of thousands of request per minute that adds up. We had a similar problem with `any?` on strings at some point